### PR TITLE
Minor frontend and backend fixes

### PR DESCRIPTION
**Pull request by smddzcy**
## About

* **Related Issues:** #124 

Various fixes for UI and backend models.

### Changes

- Fix join/leave functionality on the frontend by adding the CSRF token to the XHR request.
- Add moderators to the group model.
- Return groups of the users and moderators/members of the groups in the serializers.
- Give links to the buttons on the left nav bar (e.g. joined groups, your groups and group create button etc.)

### CODE REVIEW

File: api/interestr/api/models.py [hunk 1]
```diff
@@ -46,11 +46,11 @@ def __str__(self):
 class Group(BaseModel):
     name = models.CharField(max_length=40)
     description = models.TextField(blank=True, default='')
-    location = models.TextField(blank = True, default='')
-    tags = models.TextField(blank = True, default='') #TODO change to list
-    #tags = models.ManyToManyField(Tag, related_name='tags')
```
Comment (karacasoft) [inline api/interestr/api/models.py:51]: We should add this instead of the current one which is just a text field.

Comment (smddzcy) [reply api/interestr/api/models.py:51]: 👍

**Review by karacasoft — DISMISSED** [review]
Cool. Thanks. Approved. :+1:

**Comment by smddzcy** [comment]
@triforce930 One more commit 🤖

**Review by karacasoft — APPROVED** [review]
Approved!

**Pull request merged**